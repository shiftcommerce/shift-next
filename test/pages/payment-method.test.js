// Libraries
import Router from 'next/router'

// Pages
import PaymentMethodPage from '../../src/pages/checkout/payment-method'

jest.mock('next/config', () => () => ({
  publicRuntimeConfig: {}
}))

const cart = {
  line_items: [
    {
      id: '1'
    }
  ],
  line_items_count: 1,
  shipping_address: {},
  shipping_method: {}
}

test('renders correct checkout components', () => {
  const wrapper = shallow(
    <PaymentMethodPage cart={ cart } />,
    { disableLifecycleMethods: true }
  )

  expect(wrapper).toMatchSnapshot()
  expect(wrapper.find('PaymentMethods').length).toEqual(1)
})

describe('nextSection()', () => {
  test('navigates to shipping address when called', async () => {
    const pushSpy = jest.spyOn(Router, 'push').mockImplementation(() => {})

    const wrapper = shallow(
      <PaymentMethodPage cart={cart} />,
      { disableLifecycleMethods: true }
    )

    await wrapper.instance().nextSection()

    expect(pushSpy).toHaveBeenCalledWith('/checkout/shipping-address')

    pushSpy.mockRestore()
  })
})
