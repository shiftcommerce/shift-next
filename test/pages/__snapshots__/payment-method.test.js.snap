// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correct checkout components 1`] = `
<div>
  <PaymentMethods
    cart={
      Object {
        "line_items": Array [
          Object {
            "id": "1",
          },
        ],
        "line_items_count": 1,
        "shipping_address": Object {},
        "shipping_method": Object {},
      }
    }
    className="u-margin-top-l-des u-margin-top-l-mobtab u-margin-top-l-tab u-margin-top-l-mob"
    nextSection={[Function]}
  />
</div>
`;
